ding_user_frontend.module

```diff
@@ -27,11 +27,30 @@ function ding_user_frontend_menu_alter(&$items) {
   $provider = _ding_provider_get_provider('user');
   $items['user/%user/edit/provider_' . $provider['module']]['access'] = FALSE;
 
+  // Disable account sub-menu link (account and staff tabs).
+  $items['user/%user_category/edit/account']['access callback'] = 'ding_user_frontend_is_provider_user';
+  $items['user/%user_category/edit/account']['access arguments'] = array(1);
+
   // Change weight of user edit link, so it always second menu item.
   $items['user/%user/edit']['weight'] = -5;
 }
 
 /**
+ * Access callback to test if user is provider user.
+ *
+ * This is used to remove account and staff profile tabs on user edit pages.
+ *
+ * @param StdClass $account
+ *   User account.
+ *
+ * @return bool
+ *   If user is provider FALSE else TRUE.
+ */
+function ding_user_frontend_is_provider_user($account) {
+  return !ding_user_is_provider_user($account);
```

Comment:
i would very much like to elaborate, but have been busy. to be short i
aggree absolutely with kasper

On Tue, Jan 13, 2015 at 1:50 PM, Jesper Kristensen <notifications@github.com
> wrote:

> In ding_user_frontend.module
> <https://github.com/ding2/ding_user_frontend/pull/6#discussion-diff-22859448>
> :
>
> >    // Change weight of user edit link, so it always second menu item.
> >    $items['user/%user/edit']['weight'] = -5;
> >  }
> >
> >  /**
> > + * Access callback to test if user is provider user.
> > + *
> > + * This is used to remove account and staff profile tabs on user edit pages.
> > + *
> > + * @param StdClass $account
> > + *   User account.
> > + *
> > + * @return bool
> > + *   If user is provider FALSE else TRUE.
> > + */
> > +function ding_user_frontend_is_provider_user($account) {
> > +  return !ding_user_is_provider_user($account);
>
> Function name have been changed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ding2/ding_user_frontend/pull/6/files#r22859448>.
>